src/anyio/_core/_sockets.py

```diff
@@ -224,20 +224,23 @@ async def try_connect(remote_host: str, event: Event) -> None:
                 target_addrs.append((af, sa[0]))
 
     oserrors: list[OSError] = []
-    async with create_task_group() as tg:
-        for i, (af, addr) in enumerate(target_addrs):
-            event = Event()
-            tg.start_soon(try_connect, addr, event)
-            with move_on_after(happy_eyeballs_delay):
-                await event.wait()
-
-    if connected_stream is None:
-        cause = (
-            oserrors[0]
-            if len(oserrors) == 1
-            else ExceptionGroup("multiple connection attempts failed", oserrors)
-        )
-        raise OSError("All connection attempts failed") from cause
+    try:
+        async with create_task_group() as tg:
+            for i, (af, addr) in enumerate(target_addrs):
+                event = Event()
+                tg.start_soon(try_connect, addr, event)
+                with move_on_after(happy_eyeballs_delay):
+                    await event.wait()
+
+        if connected_stream is None:
+            cause = (
+                oserrors[0]
+                if len(oserrors) == 1
+                else ExceptionGroup("multiple connection attempts failed", oserrors)
+            )
+            raise OSError("All connection attempts failed") from cause
+    finally:
+        oserrors = []
```

Comment:
`list.clear()` is ~twice as fast as replacing it with a new list.
```suggestion
        oserrors.clear()
```